query-engine/connector-test-kit-rs/query-engine-tests/tests/new/regressions/prisma_engines_4286.rs

```diff
@@ -0,0 +1,24 @@
+use query_engine_tests::*;
+
+#[test_suite(schema(generic), only(JS, Sqlite))]
```

Comment:
JS-only because after adding the test turned out that it's actually broken with native connectors, this scenario has never been tested before (except indirectly via smoke tests for driver adapters which uncovered the issue back then).

I'll open an issue and we can fix it separately.

Reply:
The fix might be to move the logic added in this PR somewhere to the level above, let me know if you'd prefer it to be done in this PR rather than separately.